test/functional/wallet_taproot.py

```diff
@@ -499,6 +508,20 @@ def run_test(self):
             [True, False],
             lambda k1, k2: (key(k2), [multi_a(1, ([H_POINT] * rnd_pos) + [k1] + ([H_POINT] * (MAX_PUBKEYS_PER_MULTI_A - 1 - rnd_pos)))])
         )
+
+        # Test 999-of-999
+        self.do_test_k_of_n(999, 999)
+        # Test 1-of-1000, should not succeed
+        try:
+            self.do_test_k_of_n(1, 1000)
+        except JSONRPCException as e:
```

Comment:
won't this pass if no exception is raised?
Shouldn't we add something like either:
```Python
try:
    self.do_test_k_of_n(1, 1000)
    assert False, "Expected exception was not raised for 1-of-1000 test"
except JSONRPCException as e:
```

or via unittest:
```Python
# Test 1-of-1000, should not succeed
with self.assertRaises(JSONRPCException) as cm:
    self.do_test_k_of_n(1, 1000)
self.assertEqual(cm.exception.error["code"], -5)
self.assertIn("Cannot have 1000 keys in multi_a; must have between 1 and 999 keys, inclusive", cm.exception.error["message"])
```
?

Reply:
Thanks for pointing this out 🙏  I agree it shouldn't be possible for this test to succeed so an exception should be raised.

Reply:
`assert_raises_rpc_error` is probably what you want

Reply:
@paplorinc I like the second approach, it seems a bit cleaner and clearer to me. However, it doesn't seem to be used in other functional tests so maybe the first approach is better to be more consistent?

Reply:
@instagibbs Thanks! I was looking at that option too, however I couldn't see a way to use it in this context without having to change a lot of the code, but maybe I'm missing something obvious? 🤔

Reply:
if something should raise an error and a specific code/message, you use `assert_raises_rpc_error`. Just take a look on how it's used elsewhere?

Reply:
```Python
# Test that 1-of-1000 and 999-of-1000 raise exception
for k in [1, 999]:
    assert_raises_rpc_error(-5, "Cannot have 1000 keys in multi_a; must have between 1 and 999 keys, inclusive", self.do_test_k_of_n, k, 1000)
```

Reply:
I much appreciate the input @paplorinc, this was the syntax I was looking for 🙏  I saw the first version of your message and immediately thought about doing the other part ([0, 1, 999, 1000]). You read my mind, this makes very much sense! Also, doing [1, 999] would be quite time consuming because each run lasts about a second so in total it would need 1000 seconds to complete. I'll update the PR to support this version.